src/ui/components/login/login_form.rs

```diff
@@ -24,7 +26,10 @@ pub async fn login(username_or_email: String, password: String) -> Result<(), Se
     totp_2fa_token: None,
   };
 
-  let LoginResponse { jwt, .. } = client.login(req).await?;
+  let LoginResponse { jwt, .. } = client
```

Comment:
Avoid these javascripty destructurings, they're not really necessary in rust, and are really confusing to look at.

IE `let jwt = client...map_err...?.jwt`

Reply:
I can see avoiding destructuring for getting single fields, although I personally find it easier to look at when getting multiple fields, especially when they get used multiple times in the scope the destructuring occurs.